crates/lsp/src/line_index.rs

```diff
@@ -4,17 +4,9 @@
 // origin = "https://github.com/rust-lang/rust-analyzer/blob/master/crates/rust-analyzer/src/line_index.rs"
 // ---
 
-//! Enhances `ide::LineIndex` with additional info required to convert offsets
-//! into lsp positions.
-
-use settings::LineEnding;
 use triomphe::Arc;
 
-use crate::proto::PositionEncoding;
-
 #[derive(Debug, Clone)]
 pub struct LineIndex {
     pub index: Arc<biome_line_index::LineIndex>,
-    pub endings: LineEnding,
-    pub encoding: PositionEncoding,
 }
```

Comment:
Because each conversion function now takes the `biome_line_index::LineIndex`, it doesn't make sense to have a wrapper that holds these other fields too.

Instead the `Document` itself is in charge of holding these fields.

I think this makes a lot of sense, because we throw away and rebuild the `LineIndex` regularly, but the `endings` and `encoding` of the `Document` never changes and never needs to be updated.

This also becomes apparent in my next PR to optimize `on_did_change()` a bit.